CenterForOpenScience/osf.io
# [ENG-1937] Allow SHARE to view registration data

## Purpose

Move subclassed share fields to superclassed abstractprovider and ensure an access token is sent from the Registration Provider to share.

Note:

There is probably more test changes and general clean-up for this ticket then is strictly necessary, but while developing this I had a lot of difficulty with the mocking and decided to standardize the mocking to use a `responses` fixture and run celery live to ensure test were comprehensive. I have listed these changes under "Non-Functional requirements" in changes.


## Changes

Functional requirements:
- moves field location to super model and add migration
- Sends SHARE data from registration provider.


Non-Functional requirements:
- Makes all SHARE usage configurable so it's easily disabled/enabled for local development via the SHARE_ENABLED setting.
- DRYs up many SHARE functions and makes a pytest fixture that mocks SHARE is a more reliable way.
- Moves share specific tests into their own directory and removes many patches and mocks.
- `share_publish_type` is now takes advantage of the provider standardization and no longer needs to be passed specially through many functions.


## QA Notes

This effects the AbstractProvider Model and Provider API Endpoints

- Does this change require a data migration? Yes a small one
- What is the level of risk? Low
- Any permissions code touched? No

 QA can verify this via the admin app.

## Documentation

technical task, no new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1937
